FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,45 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			
+			switch result {
+			case let .success((data, response)):
+				if response.statusCode == 200,
+				   let root = try? JSONDecoder().decode(Root.self, from: data) {
+					completion(.success(root.mapToFeedImages()))
+				} else {
+					completion(.failure(Error.invalidData))
+				}
+			default:
+				completion(.failure(Error.connectivity))
+			}
+		}
+	}
+
+	private struct Root: Decodable {
+		private let items: [ImageItem]
+
+		private struct ImageItem: Decodable {
+			let image_id: UUID
+			let image_desc: String?
+			let image_loc: String?
+			let image_url: URL
+		}
+
+		func mapToFeedImages() -> [FeedImage] {
+			var result = [FeedImage]()
+			for item in items {
+				let feedImage = FeedImage(
+					id: item.image_id,
+					description: item.image_desc,
+					location: item.image_loc,
+					url: item.image_url
+				)
+				result.append(feedImage)
+			}
+			return result
```

Comment:
This part can be written in a functional way which is slightly shorter and more concise

No need to make this amendment, it is just for your interest.

```suggestion
			return items.map({ item in
				let feedImage = FeedImage(
					id: item.image_id,
					description: item.image_desc,
					location: item.image_loc,
					url: item.image_url
				)
				result.append(feedImage)
			})
```